dist/tools/headerguards/check.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+# Copyright 2017 Oliver Hahm <oliver.hahm@inria.fr>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+# customizable
+DIRS="core drivers pkg sys"
```

Comment:
why not `boards`? If we do the proposed `vendor/` include structure, we might also be able to add `cpu`.